### PR TITLE
Adds in-emacs editing to spacemacs/report-issue

### LINKED ISSUE #5902: spacemacs/report-issue fails poorly if I'm not logged into github
#### Description
spacemacs/report-issue appears to assume that I'm running emacs on a machine whose default browser is logged into github.  This isn't true for me, which is fine.  But the failure mode is a generic github page that says "this page took way too long to load", which is hardly helpful.

If it's not possible to fail in a less misleading way, could we do away with having spacemacs/report-issue open the browser?  I'd be happy to copy-paste this template (with the system info filled in) out of a buffer into a github form that I've opened manually.

#### Reproduction guide
- Start Emacs on a box where your default browser isn't logged into github
- Call spacemacs/report-issue

*Observed behaviour:*
Browser opens a github error page that says, "This page took way too long to load."

*Expected behaviour:*
Browser opens with the github page for creating a new issue, or at least a github login page.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.2
- Spacemacs: 0.105.13
- Spacemacs branch: chrome-layer-plain-text-gmail (rev. 107b53f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
((auto-completion :variables auto-completion-return-key-behavior 'complete auto-completion-tab-key-behavior 'cycle auto-completion-enable-help-tooltip t)
 (chrome :variables chrome-gmail-rich-text-format nil)
 emacs-lisp
 (jane :variables jane-jane-elisp-path "/usr/local/home/dwang/workspaces/jane/emacs/dev/spacemacs/jane-tweaks/rewrite/ share /app/emacs/" Omake\.Server\.program "/j/office/app/emacs/prod/jane-elisp/bin/build-manager")
 markdown org sql
 (version-control :variables version-control-diff-tool 'diff-hl)
 vinegar w3m)
```

#### Backtrace
```

```

### PR DESCRIPTION
**Pull request by dwang20151005**
spacemacs/report-issue opens a buffer prepopulated with a template and some
system information. The buffer is markdown-mode augmented with a keybinding to
send the contents to github.

Or if that doesn't work, the text remains in the buffer to be manually copied
and pasted. This makes the error reporting more robust: right after the user has
experienced one bug is the worst time to manifest another one.

This commit also rewrites spacemacs//describe-last-keys-string because I
couldn't figure out how else to prevent view-lossage from burying the new
REPORT_SPACEMACS_ISSUE buffer.  Suggestions welcome.

Fixes #5902 , supercedes #5901

### CODE REVIEW

File: core/core-spacemacs.el [hunk 1]
```diff
@@ -249,42 +255,63 @@ defer call using `spacemacs-post-user-config-hook'."
...
+that the issue has been created successfully, you can close this buffer.
+
+Markdown syntax is supported in this buffer.
+
+\\{spacemacs/report-issue-mode-map}
+")
+
+(define-key
+  spacemacs/report-issue-mode-map
+  (kbd "C-c C-c")
```
Comment (StreakyCobra) [inline core/core-spacemacs.el]: https://github.com/syl20bnr/spacemacs/blob/develop/doc/CONVENTIONS.org#confirm-and-abort

**Comment by StreakyCobra** [comment]
Seems nice, I just had a quick review.

File: core/core-spacemacs.el (hunk 1 above)
Comment (dwang20151005) [inline core/core-spacemacs.el]: Thanks for catching that.

**Comment by dwang20151005** [comment]
I see that I broke something.  Looking into it now.

**Comment by robbyoconnor** [comment]
<3

**Comment by syl20bnr** [comment]
Nice 💜 
I added some font lock to colorize the expression to replace.
Thank you ! 👍 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**